ruth/data/map.py

```diff
@@ -65,11 +88,48 @@ def name(self):
         """Name of the map."""
         return self.border.name
 
+    def init_current_speeds(self):
+        speeds = nx.get_edge_attributes(self.network, name='speed_kph')
+        lengths = nx.get_edge_attributes(self.network, name='length')
+        travel_times = {}
+        for key, value in speeds.items():
+            travel_times[key] = float(lengths[key]) / float(value)
+        nx.set_edge_attributes(self.network, values=speeds, name="current_speed")
+        nx.set_edge_attributes(self.network, values=travel_times, name="current_travel_time")
+
+    def update_current_speeds(self, segment_ids, speeds):
+        new_speeds = {}
+        new_travel_times = {}
+        lengths = nx.get_edge_attributes(self.simple_network, name='length')
+        for (node_from, node_to), speed in zip(segment_ids, speeds):
+            new_speeds[(node_from, node_to)] = speed
+            new_travel_times[(node_from, node_to)] = float(lengths[(node_from, node_to)]) / float(speed)
+        nx.set_edge_attributes(self.simple_network, values=new_speeds, name='current_speed')
```

Comment:
The current speed will be useful for the cpp/distributed alternatives.

Reply:
Hmm. Does that mean that the cpp version computes its own edge labels by dividing the segment length by `current_speed`? That sounds a bit scary, as this logic might not be identical in the simulator and in alternatives.

If distributed alternatives are used, then we don't need the travel times, right? Because we don't even compute any shortest paths locally. Maybe we should somehow detect this, because it's needless work to compute it if we don't use it. At the very least, it would be nice to measure how long does it take to run this function :) If it's quick, then we don't have to worry about it.

Reply:
Yeah, cpp probably computes the edge weights by itself. One difference is for example that it ignores edges with speed 0 (which turned out to be a bit issue for the version with updated map, but we are working on it).

Based on some py-spy measurements we took the update_current_speed function takes about 2 seconds in total, so it's not that bad.

Reply:
2 seconds for how many edges updated?

Reply:
Cannot say rn, but 2 seconds for cca 45 minutes of simulation running, if I understand the py-spy record correctly and the map being updated every next step of the simulation. The most time consuming in the simulation rn is computing the alternatives (then saving the map and the simulation I think).

Reply:
Sorry, I missed the `in total` part. Then it should be fine.